src/GraphsDFG/services/GraphsDFG.jl

```diff
@@ -334,7 +334,7 @@ Optionally provide a distance to specify the number of edges should be followed.
 Optionally provide an existing subgraph addToDFG, the extracted nodes will be copied into this graph. By default a new subgraph will be created.
 Note: By default orphaned factors (where the subgraph does not contain all the related variables) are not returned. Set includeOrphanFactors to return the orphans irrespective of whether the subgraph contains all the variables.
 """
-function getSubgraphAroundNode(dfg::GraphsDFG{P}, node::T, distance::Int64=1, includeOrphanFactors::Bool=false, addToDFG::GraphsDFG=GraphsDFG{P}(); solvable::Int=0)::GraphsDFG where {P <: AbstractParams, T <: DFGNode}
+function getSubgraphAroundNode(dfg::GraphsDFG{P}, node::T, distance::Int64=1, includeOrphanFactors::Bool=false, addToDFG::G=GraphsDFG{P}(); solvable::Int=0)::AbstractDFG where {P <: AbstractParams, T <: DFGNode, G <: AbstractDFG}
```

Comment:
With the new API, the verb would probably be something like deepcopy.

Reply:
Agreed, that is important for sure

Reply:
With this you should be able to pass an destination graph and it should add to it, rather than assume a return of a GraphsDFG. 

Above, are we talking about renaming this to deepcopy?

Reply:
With new API `get` returns by reference if possible. So `build` or `deepcopy` fits better.
getSubgraph sounds like it should return a 'view'

Reply:
I added `view` to the api discussion. Would be cool to implement something like `viewSubGraph` for the clique subgraphs.